File: keeprainy/leetcode/weekly_contest_265/2057_Smallest_Index_With_Equal_Value.py
"""
#  LeetCode
#  ver.Python3
#
#  Created by GGlifer
#
#  Open Source
"""

from typing import *


class Solution:
    def smallestEqual(self, nums: List[int]) -> int:
        for i, n in enumerate(nums):
            if i % 10 == n:
                return i
        return -1


def main():
    solution = Solution()

    nums = [0, 1, 2]

    print(solution.smallestEqual(nums))


if __name__ == '__main__':
    main()

File: sally/leetcode/weekly-contest-265/2057. Smallest Index With Equal Value.cpp
class Solution {
public:
    int smallestEqual(vector<int>& nums) {
        int num_len = nums.size();
        for(int i = 0 ; i < num_len; i++)
            if(i % 10 == nums[i])
                return i;
        return -1;
    }
};